src/tools/components/AlignOverview.tsx

```diff
@@ -0,0 +1,225 @@
+  conservationOptions: ConservationOptions;
+  totalLength: number;
+  annotation: FeatureType | undefined;
+  selectedId?: string;
+  setSelectedId: Dispatch<SetStateAction<string | undefined>>;
+};
+
+const AlignOverview: FC<BlastOverviewProps> = ({
+  alignment,
+  alignmentLength,
+  highlightProperty,
+  conservationOptions,
+  totalLength,
+  annotation,
+}) => {
+  const [highlightPosition, setHighlighPosition] = useState('');
+  const [initialDisplayEnd, setInitialDisplayEnd] = useState<
+    number | undefined
+  >();
+  const [displayEnd, setDisplayEnd] = useState<number>();
+  const [msaOffsetTop, setMsaOffsetTop] = useState<number | undefined>();
+
+  const tracksOffset = Math.max(...alignment.map(({ from }) => from));
+
+  const findHighlighPositions = useCallback(
+    ({ displaystart, displayend }: EventDetail) => {
+      const displayStart = parseInt(displaystart, 10);
+      const displayEnd = parseInt(displayend, 10);
+      const start = tracksOffset + displayStart;
+      const end = tracksOffset + displayEnd;
+      setHighlighPosition(`${start}:${end}`);
+      setDisplayEnd(displayEnd);
+    },
+    [tracksOffset]
+  );
+
+  const managerRef = useCallback(
+    (node): void => {
+      if (node && initialDisplayEnd) {
+        node.addEventListener('change', ({ detail }: { detail: EventDetail }) =>
+          findHighlighPositions(detail)
+        );
+        node.setAttribute('displaystart', 1);
+        node.setAttribute('displayend', initialDisplayEnd);
+        setHighlighPosition(
+          `${tracksOffset}:${tracksOffset + initialDisplayEnd}`
+        );
+      }
+    },
+    [initialDisplayEnd, findHighlighPositions, tracksOffset]
+  );
+
+  const setMSAAttributes = useCallback(
+    (node): void => {
+      if (!node) {
+        return;
+      }
+      setMsaOffsetTop(node.offsetTop);
+
+      const displayEndValue =
+        alignmentLength / (15 / node.getSingleBaseWidth());
+
+      const maxSequenceLength = Math.max(
+        ...alignment.map((al) => al.sequence.length)
+      );
+      if (displayEndValue < maxSequenceLength) {
+        setInitialDisplayEnd(displayEndValue);
+      } else {
+        setInitialDisplayEnd(maxSequenceLength);
+      }
+
+      node.data = alignment.map(({ name, sequence }) => ({ name, sequence }));
+    },
+    [alignment, alignmentLength]
+  );
+
+  // This should use state to handle selection of alignment and set features
+  const features = useMemo(() => alignment[1].features, [alignment]);
+
+  const setFeatureTrackData = useCallback(
+    (node): void => {
+      if (node && features && annotation) {
+        let processedFeatures = processFeaturesData(
+          features.filter(({ type }) => type === annotation)
+        );
+        processedFeatures = transformFeaturesPositions(processedFeatures);
+        node.data = processedFeatures;
+      }
+    },
+    [features, annotation]
+  );
+
+  const overviewHeight: string = (alignment && alignment.length > 10
```

Comment:
Not sure why this needs to casted into a string, as later it is just parsed to a number again with the component that will use it

Reply:
I think this was an accident.

Reply:
Fixed.